gateway/src/apicast/mapping_rule.lua

```diff
@@ -123,7 +124,8 @@ function _M.from_proxy_rule(proxy_rule)
     proxy_rule.querystring_parameters,
     proxy_rule.metric_system_name,
     proxy_rule.delta,
-    proxy_rule.last
+    proxy_rule.last,
+    tonumber(proxy_rule.owner_id)
```

Comment:
If you really want to use `owner_id` here instead of the proposed `backend_id`, then you will also need `owner_type`. That's because the proxy rule can belong to a product (`owner_type == "Proxy"`) or to a backend (`owner_type == "BackendApi"`). In the first case, it should be considered for the match despite `owner_id`, while in the latter case, only if `owner_id` matches the one of the routing.